Add unit tests for com.jme3.renderer.Camera

sudo-jaa opened the pull request:
Also added additional` .equals()` methods for Camera.java and Plane.java to assist testing of the Camera class. 

This PR contains a suite of tests for `com.jme3.renderer.Camera`, once again utilising [Diffblue](https://www.diffblue.com/) [Cover](https://www.diffblue.com/datasheet) to assist with the testing workflow. The rest of the requested classes and packages are on my radar; i'll be looking into them next.

freddytuxworth commented on jme3-core/src/main/java/com/jme3/math/Plane.java:
```diff
@@ -291,4 +292,13 @@ public Plane clone() {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Plane plane = (Plane) o;
+        return Float.compare(plane.constant, constant) == 0 &&
```
replace `Float.compare(...)` with `plane.constant == constant`

freddytuxworth commented on jme3-core/src/main/java/com/jme3/math/Plane.java:
```diff
@@ -33,6 +33,7 @@
 
 import com.jme3.export.*;
 import java.io.IOException;
+import java.util.Objects;
```
Is this used?

freddytuxworth commented on jme3-core/src/main/java/com/jme3/renderer/Camera.java:
```diff
@@ -999,6 +1001,7 @@ public float distanceToNearPlane(Vector3f pos) {
         return worldPlane[NEAR_PLANE].pseudoDistance(pos);
     }
 
+
```
remove added newline

freddytuxworth commented on jme3-core/src/test/java/com/jme3/renderer/CameraTest.java:
```diff
@@ -0,0 +1,276 @@
+        baseCamera.setViewPort(1, 2, 3, 4);
+        baseCamera.setFrustumPerspective(1.0f, 2.5f, 6.5f, 8.5f);
+
+        assertNotEquals(baseCamera, copyToCamera);
```
Why?

freddytuxworth commented on jme3-core/src/test/java/com/jme3/renderer/CameraTest.java:
```diff
@@ -0,0 +1,276 @@
+        assertEquals(0.5, camera.frustumRight, 0.0f);
+        assertEquals(0.5, camera.frustumTop, 0.0f);
+        assertEquals(-0.5, camera.frustumBottom, 0.0f);
+
```
replace all `.frustum` with `.getFrustum`

sudo-jaa replied:
Repo owner prefer using direct field access over using the getters. I'm not sure why, exactly.

freddytuxworth commented on jme3-core/src/test/java/com/jme3/renderer/CameraTest.java:
```diff
@@ -0,0 +1,276 @@
+        assertEquals(camera.getWidth(), 2);
+        assertEquals(camera.getHeight(), 3);
+
+        camera.resize(2, 3, true);
```
If you use `resize(2, 4, true)`, do the `0.333`s become `0.5`s and can you drop the `delta` to 0?

freddytuxworth commented on jme3-core/src/test/java/com/jme3/renderer/CameraTest.java:
```diff
@@ -0,0 +1,276 @@
+        final Vector3f worldUpVector = new Vector3f(30, 40, 50);
+        final Camera camera = new Camera(1, 1);
+
+        assertEquals(new Quaternion(0.0f, 0.0f, 0.0f, 1.0f), camera.getRotation());
```
Remove assertion about rotation before `lookAt` is called, because the default might change

freddytuxworth commented on jme3-core/src/test/java/com/jme3/renderer/CameraTest.java:
```diff
@@ -0,0 +1,276 @@
+        final Vector3f worldUpVector = new Vector3f(0, 0, 0);
+        final Camera camera = new Camera(1, 1);
+
+        assertEquals(new Quaternion(0.0f, 0.0f, 0.0f, 1.0f), camera.getRotation());
```
Remove assertion about rotation before `lookAt` is called, because the default might change

freddytuxworth commented on jme3-core/src/test/java/com/jme3/renderer/CameraTest.java:
```diff
@@ -0,0 +1,276 @@
+        assertTrue(camera.containsGui(bounds1));
+
+        final BoundingVolume bounds2 = new BoundingSphere(-50.0f, new Vector3f(20.0f, 10.0f, 3.0f));
+        assertTrue(camera.containsGui(bounds2));
```
a test returning false would be good

Commit:
Add unit tests for com.jme3.renderer.Camera
    Also add additional .equals() methods for Camera.java and Plane.java to assist testing of the Camera class

peterschrammel commented on jme3-core/src/main/java/com/jme3/math/Plane.java:
```diff
@@ -291,4 +291,13 @@ public Plane clone() {
         }
     }
+
+    @Override
```
Put changes to this file into a separate commit to separate from the tests

peterschrammel commented on jme3-core/src/main/java/com/jme3/renderer/Camera.java:
```diff
@@ -1480,4 +1482,38 @@ public void read(JmeImporter e) throws IOException {
         onViewPortChange();
         onFrameChange();
     }
+
```
Put changes to this file into a separate commit to separate from the tests
